platform/darwin/src/MGLMapSnapshotter.mm

```diff
@@ -292,7 +321,23 @@ + (MGLImage*)drawAttributedSnapshotWorker:(mbgl::MapSnapshotter::Attributions)at
     UIGraphicsBeginImageContextWithOptions(mglImage.size, NO, scale);
     
     [mglImage drawInRect:CGRectMake(0, 0, mglImage.size.width, mglImage.size.height)];
-    
+
+    CGContextRef currentContext = UIGraphicsGetCurrentContext();
+
+    if (currentContext && overlayHandler) {
+        MGLMapSnapshotOverlay *snapshotOverlay = [[MGLMapSnapshotOverlay alloc] initWithContext:currentContext];
+        CGContextSaveGState(snapshotOverlay.context);
+        overlayHandler(snapshotOverlay);
+        CGContextRestoreGState(snapshotOverlay.context);
+        currentContext = UIGraphicsGetCurrentContext();
+    }
+
+    if (!currentContext && overlayHandler) {
```

Comment:
It doesn’t seem possible for the developer to nil out `currentContext`. `CGContextRef` is a typedef for `CGContext *`. What we pass to the MGLMapSnapshotOverlay initializer is a `CGContext *` pointer rather than a reference such as `CGContext &`, so even if the developer were to somehow nil out the overlay object’s pointer, the `currentContext` local variable would continue to point to the same object it did initially.

Reply:
Oh I see, this is to account for the edge case where there’s no current context, just like the `if` block above. 👍